### PR TITLE
[kealib] Cleanup, fix CMake config

### PR DESCRIPTION
**Pull request by dg0yt**
Cherry-picked from #35437.

### CODE REVIEW

File: ports/kealib/portfile.cmake [hunk 1]
```diff
@@ -4,7 +4,9 @@ vcpkg_from_github(
     REF "kealib-${VERSION}"
     SHA512 82399f1332ff2aeb6342732e9e5c897c813109fd18e77cfc8d866f06adf4faa7f080f1f3c0a3b777fb3a679912dacf4851b7ad09a338d6087dd1d26eb2d1689f
     HEAD_REF master
-    PATCHES hdf5_include.patch
+    PATCHES
+        kealib-target.diff
```
Comment (dg0yt) [inline ports/kealib/portfile.cmake:8]: https://github.com/ubarsc/kealib/pull/45

File: ports/kealib/portfile.cmake [hunk 2]
```diff
@@ -4,7 +4,9 @@ vcpkg_from_github(
     REF "kealib-${VERSION}"
     SHA512 82399f1332ff2aeb6342732e9e5c897c813109fd18e77cfc8d866f06adf4faa7f080f1f3c0a3b777fb3a679912dacf4851b7ad09a338d6087dd1d26eb2d1689f
     HEAD_REF master
-    PATCHES hdf5_include.patch
```
Comment (dg0yt) [inline ports/kealib/portfile.cmake:7]: Pointless patch.

**Comment by MonicaLiu0311** [comment]
When testing usage, the following error occurs:
```
1> [CMake] CMake Error at G:/kealib/installed/x64-windows/share/libkea/libkeaTargets.cmake:60 (set_target_properties):
1> [CMake]   The link interface of target "libkea::kea" contains:
1> [CMake] 
1> [CMake]     hdf5::hdf5-shared
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   G:/kealib/installed/x64-windows/share/libkea/libkeaConfig.cmake:27 (include)
1> [CMake]   G:/kealib/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
1> [CMake]   CMakeUsage/CMakeLists.txt:14 (find_package)
1> [CMake] -- Generating done (0.1s)
1> [CMake] CMake Generate step failed.  Build files cannot be regenerated correctly.
```
<details>
<summary><b>test.cpp</b></summary>
<pre>
#include &#60;iostream&#62;
#include "libkea/KEACommon.h"

using namespace std;

int main()
{
        cout << "Hello CMake." << endl;
        return 0;
}
</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

set(CMAKE_TOOLCHAIN_FILE "/home/monica/vcpkg/scripts/buildsystems/vcpkg.cmake")

project ("test")

add_library (main "test.cpp")

find_package(libkea CONFIG REQUIRED)
target_link_libraries(main PRIVATE libkea::Kealib)
</pre></details>

**Comment by dg0yt** [comment]
@MonicaLiu0311 Please review again.

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found).

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Pull request merged**